dipy/viz/horizon/util.py

```diff
@@ -165,14 +165,18 @@ def is_binary_image(data, unique_points=100):
     boolean
         Whether the image is binary or not
     """
-    indices = []
 
-    rng = np.random.default_rng()
+    sample_cube = [sample_cube_size] * data.ndim
 
-    for dim in data.shape:
-        indices.append(rng.integers(0, dim - 1, size=unique_points))
+    for idx, dim in enumerate(data.shape):
+        if dim < sample_cube[idx]:
+            data = np.take(data, np.arange(stop=dim), axis=idx)
+        else:
+            start = int(dim/2) - int(sample_cube[idx]/2)
+            stop = int(dim/2) + int(sample_cube[idx]/2)
+            data = np.take(data, np.arange(start=start, stop=stop), axis=idx)
 
-    return np.unique(np.take(data, indices)).shape[0] <= 2
+    return np.unique(data).shape[0] <= 2
```

Comment:
Why not `==2`?  

Not sure if having 1 values or 0 values can be considered binary

Reply:
This is in case of we do not find multiple values in the small box. This is how the old logic was, so made sure to keep it the same.

Reply:
ok